aya/src/programs/xdp.rs

```diff
@@ -72,6 +74,7 @@ impl Xdp {
     ///
     /// See also [`Program::load`](crate::programs::Program::load).
     pub fn load(&mut self) -> Result<(), ProgramError> {
+        self.data.expected_attach_type = Some(bpf_attach_type::BPF_XDP);
```

Comment:
do we actually need to set this?

Reply:
it does work fine without it, but setting it doesn't hurt. especially as libbpf does.
I noticed it comparing the strace of aya vs libbpf.